pandas/io/parsers/base_parser.py

```diff
@@ -549,7 +558,7 @@ def _convert_to_ndarrays(
         return result
 
     def _set_noconvert_dtype_columns(
-        self, col_indices: List[int], names: List[Union[int, str]]
+        self, col_indices: List[int], names: List[int | str | tuple]
```

Comment:
see #39538 for some discussion on this syntax change - perhaps it's best to defer it?

Reply:
Rolled back